src/operator.rs

```diff
@@ -52,7 +42,14 @@ pub trait EvaluatableBorrowed<T> {
 }
 
 pub trait UnaryPreAggOperator: Send + Sync {
-    fn process(&self, rec: Record) -> Result<Option<Record>, EvalError>;
+    /// Returns true if the operator is still willing to accept input records.
+    fn is_open(&self) -> bool { true }
```

Comment:
In the current structure, an operator can only return one record.  For something like `limit 10` to work correctly, it would need to output the tenth record and then the EndOfStream record without waiting to be called with another record.  I think it would be good to allow operators to output more than one record, but I didn't want to tackle that yet.

Reply:
ahh right I forgot about that! yeah I think drain is the best option.